simulate-event.js

```diff
@@ -272,6 +292,9 @@ module.exports = function (element, type, options) {
     cancelable: true
   }, result(eventOptions, eventType, element, type, options), options);
 
+  // Preserve the original or use the new options if none were given.
+  if (!original) original = options;
```

Comment:
Should we maybe create a function to pull off non-enumerable properties when we're under certain keyboard events? I think that would make sense, basically what you have but return it as an object so it'll be merged with options still.